src/components/pageData.js

```diff
@@ -0,0 +1,162 @@
+import http from 'http'
+import https from 'https'
+import clone from 'clone'
+import pageNameChanger from './pageNameChanger'
+import pageNameSanitiser from './pageNameSanitiser'
+
+const PageData = class PageData {
+  constructor (params = { }) {
+    this.params = params
+    return this.init()
+  }
+
+  returnFiles () {
+    return this.params.files
+  }
+
+  extractData (data, fileName, fileParams) {
+    const { extraFiles, dataSource = { } } = fileParams
+    const { repeater, pageDataField, pageNameField } = dataSource
+    const newFiles = [ ]
+    if (repeater) data = data[repeater]
+    if (!data || !data.length) {
+      return {
+        data,
+        response: newFiles
+      }
+    }
+    for (let i = 0; i < data.length; i++) {
+      let newFile = extraFiles ? { } : clone(this.params.files[ fileName ])
+      let folder = fileName.split('/')
+      folder = (folder && folder.length > 1) ? folder.slice(0, -1).join('/') + '/' : ''
+
+      // Attach page data to props passed to the page
+      let pageData = data[i]
```

Comment:
Are we extending the current pageData object for ease in the template? Could we get any benefits from attaching a new object?